content/posts/2021/07-15-salt-nornir-intro.md

```diff
@@ -0,0 +1,229 @@
+# SALTSTACK Nornir Proxy Minion introduction
+
+Network Automation is a complex but interesting topic with a lot of problems to solve and a lot to learn, defined by one of the [vendors](https://www.cisco.com/c/en/us/solutions/automation/network-automation.html) as:
+
+> Network automation is the process of automating the configuring, managing, testing, deploying, and operating of physical and virtual devices within a network.
+
+This post is an introduction in what SALTSTACK is and what is Nornir and how Network Automation field can benefit from combining them together.
+
+## Nornir ~~or~~ and SALTSTACK
+
+Many systems were developed to address automation aspects of networking, Nornir and SALTSTACK falls in same category and could be briefly introduced as:
+
+**Nornir** - *is an automation framework written in python to be used with python* - it is a general purpose, automation system that uses plugins to address specific problems at specific layers.
+
+**SALTSTACK** - *Built on python, Salt uses simple and human-readable YAML combined with event-driven automation to deploy and configure complex IT systems* - it is a general purpose, hierarchical, automation system that uses modules to address specific problems at specific levels.
+
+Both systems share many common aspects, such as:
+
+- Both written in Python and can be directly interfaced using Python API
+- Both use plugins/modules to interact with devices and external systems
+- Both are open-source (SALTSTACK community) and available on GitHub
+- Both follow *"everything is pluggable and extendable"* paradigm
+
+But, while Nornir targets mainly automation of netowrks, SALTSTACK was initially developed to automate IT systems - servers, virtual machines, operating systems, applications.
+
+SALT also aims to address a much wider set of use cases compared to Nornir, it has more components, longer history (111,991 commits) and big community. Moreover, SALTSTACK has a command of developers backing it up and was recently [purchased](https://blogs.vmware.com/management/2020/10/vmware-completes-saltstack-acquisition-to-bolster-software-configuration-management-and-infrastructure-automation.html) by VMWare. Worth noting that SALTSTACK has Enterprise version as well.
```

Comment:
"SALTSTACK has a command of developers"

I think command may be a typo? Not sure what you mean with this.

"Worth noting that SALTSTACK has AN Enterprise version as well"